src/GenSequence.ts

```diff
@@ -27,6 +27,7 @@ export interface Sequence<T> extends IterableLike<T> {
     any(fnFilter: (t: T)=> boolean): boolean;
     first(fnFilter?: (t: T)=> boolean, defaultValue?: T): Maybe<T>;
     first(fnFilter: (t: T)=> boolean, defaultValue: T): T;
+    max(): Maybe<T>;
```

Comment:
When the list is small, the order doesn't really matter.  As you pointed it out, as it grows, a logical order become necessary.

I would say group by type: Reducers, Filters, Mappers; and then alphabetically.